src/cpp/core/include/core/r_util/RActiveSession.hpp

```diff
@@ -0,0 +1,492 @@
+/*
+ * RActiveSession.hpp
+ *
+ * Copyright (C) 2022 by RStudio, PBC
+ *
+ * Unless you have received this program directly from RStudio pursuant
+ * to the terms of a commercial license agreement with RStudio, then
+ * this program is licensed to you under the terms of version 3 of the
+ * GNU Affero General Public License. This program is distributed WITHOUT
+ * ANY EXPRESS OR IMPLIED WARRANTY, INCLUDING THOSE OF NON-INFRINGEMENT,
+ * MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE. Please refer to the
+ * AGPL (http://www.gnu.org/licenses/agpl-3.0.txt) for more details.
+ *
+ */
+
+
+#ifndef CORE_R_UTIL_ACTIVE_SESSION_HPP
+#define CORE_R_UTIL_ACTIVE_SESSION_HPP
+
+#include <boost/noncopyable.hpp>
+
+#include <core/Log.hpp>
+#include <core/Settings.hpp>
+#include <core/DateTime.hpp>
+
+#include <core/r_util/RSessionContext.hpp>
+#include <core/r_util/RProjectFile.hpp>
+#include <core/r_util/RActiveSessionStorage.hpp>
+
+#include <shared_core/Error.hpp>
+#include <shared_core/FilePath.hpp>
+#include <shared_core/SafeConvert.hpp>
+#include <shared_core/json/Json.hpp>
+
+namespace rstudio {
+namespace core {
+namespace r_util {
+
+class ActiveSession : boost::noncopyable
+{
+public:
+   explicit ActiveSession(const std::string& id) : id_(id) 
+   {
+   }
+
+   explicit ActiveSession(
+      const std::string& id,
+      const FilePath& scratchPath,
+      std::shared_ptr<IActiveSessionStorage> storage) : id_(id), scratchPath_(scratchPath), storage_(storage)
+   {
+   }
+
+   const std::string kExecuting = "executing";
+   const std::string kInitial = "initial";
+   const std::string kLabel = "label";
+   const std::string kLastUsed = "last_used";
+   const std::string kProject = "project";
+   const std::string kSavePromptRequired = "save_prompt_required";
+   const std::string kRunning = "running";
+   const std::string kRVersion = "r_version";
+   const std::string kRVersionHome = "r_version_home";
+   const std::string kRVersionLabel = "r_version_label";
+   const std::string kWorkingDir = "working_dir";
```

Comment:
So this used to be `working_directory`, not `working_dir` but doesn't that mean we won't be able to read old suspended sessions after an upgrade?  Same thing with `launch_params`.  I don't know if the clean up is worth the aggravation that will cause.  Also, we also use `working_directory` in settings so maybe we should just go back to using the full name to be consistent?  Same with `launch_parameters`.

Reply:
Never mind, I see the file name is actually working-dir that comes from the mapping table.